loadgen/kitchen_sink_executor.go

```diff
@@ -30,18 +30,30 @@ func (k KitchenSinkExecutor) Run(ctx context.Context, options RunOptions) error
 		}
 	}
 	// Create generic executor and run it
-	return GenericExecutor{
+	ge := &GenericExecutor[any]{
 		DefaultConfiguration: k.DefaultConfiguration,
-		Execute: func(ctx context.Context, run *Run) error {
+		Execute: func(ctx context.Context, run *Run, _ *any) error {
 			options := run.DefaultKitchenSinkWorkflowOptions()
 			// Shallow copies params, users are expected not to mutate any slices
 			options.Params = params
 			if k.UpdateWorkflowOptions != nil {
-				k.UpdateWorkflowOptions(ctx, run, &options)
+				err := k.UpdateWorkflowOptions(ctx, run, &options)
+				if err != nil {
+					return err
+				}
 			}
 			return run.ExecuteKitchenSinkWorkflow(ctx, &options)
 		},
-	}.Run(ctx, options)
+	}
+	return ge.Run(ctx, options)
+}
+
+func (k KitchenSinkExecutor) PreScenario(_ context.Context, _ ScenarioInfo) error {
```

Comment:
I don't think we should make users implement pre/post scenario. I think they can have a run, and if they want to do something at the beginning or end, it's their code to do so as they please